### ISSUE TITLE
Tooltip tour should only appear when clicking on the "Update dashboard" CTA.

### ISSUE THREAD

**Issue by bethanylang** [google/site-kit-wp#6914]
## Bug Description

As reported by @wpdarren in Bug Bash:

First time setting up Site Kit and creating a new GA4 account/property and when you're redirected back to the dashboard, the tooltip tour appears. I wouldn't expect this to appear when it's the first time setting up Site Kit.    

## Steps to reproduce

1. Create a new test site
2. Set up Site Kit with Analytics for the first time with GA4 Reporting and GA4 Activation Banner flags activated
3. Go to Site Kit > Dashboard and feature tour will appear

---------------

_Do not alter or remove anything below. The following sections will be managed by moderators only._

## Acceptance criteria

- The conditions for showing the GA4 Reporting feature tour should be updated so it is only triggered when the user clicks the ["Update dashboard" CTA](https://www.figma.com/file/0UCIE3hPeiu4NYdpAODASd/GA4-Updates?node-id=826-7063&t=wsvpNXFVSYAO2IDc-4) and their dashboard is reloaded with GA4 stats.

Note: the feature tour will also be triggered when the dashboard is automatically switched over to GA4, [this banner](https://www.figma.com/file/0UCIE3hPeiu4NYdpAODASd/GA4-Updates?node-id=581-8816&t=wsvpNXFVSYAO2IDc-4) is displayed, and the user clicks the "Learn what's new" CTA, however this is outside the scope of this issue and will be addressed in https://github.com/google/site-kit-wp/issues/6558 / https://github.com/google/site-kit-wp/issues/6549.

## Implementation Brief

Within `assets/js/modules/search-console/components/dashboard/SearchFunnelWidgetGA4/Overview/index.js`:

- Remove the `useEffect()` that triggers the `triggerOnDemandTour()` call that triggers the GA4 Reporting feature tour.

Within `assets/js/components/notifications/SwitchGA4DashboardViewNotification.js`:

- Update `handleCTAClick()` to include the `triggerOnDemandTour()` call and the `setValue()` call that sets the `forceInView` value that were removed from the `SearchFunnelWidgetGA4/Overview` component.

### Test Coverage

* No new tests needed.

## QA Brief

* Set up a new site and connect the Analytics module, but select the universal analytics dashboard view.
* Activate the `ga4Reporting` feature flag
* Go to the dashboard and ensure that use see the banner that suggest you switching to the GA4 dashboard view.
* Click on the CTA button after which in a short period of time the dashboard should switch to GA4 and the GA4 reporting feature tour should start.

## Changelog entry

* Update GA4 dashboard tour to only be triggered by banner CTA.

**Comment by techanvil** [comment]
Hi @marrmarr, having started looking into the IB for this I've realised there are technical challenges making the AC as defined impossible to fully achieve.

> With `ga4Reporting` enabled, when Analytics is connected for the first time and the user is redirected back to the Dashboard, the GA4 Reporting feature tour should not be triggered.

I initially specced the AC as quoted above, to say the tour should not be triggered the first time a site sets up Analytics, as the general case seemed logical (as opposed to "when the plugin is first setup and Analytics is connected"). However it turns out this is technically complicated and indeed not possible to fully implement as we have no historical record of when an existing site has previously connected Analytics and then disconnected it, so we can't reliably track when an existing site sets up Analytics for the first time.

We could tighten up the AC to be more specific, i.e. the case when the plugin is first setup (assuming it's a version which includes this issue) and Analytics is connected - we could manage this by introducing some new state to track when Site Kit and Analytics are setup and so forth. This does add a bit of complexity but it should be achievable and is also more aligned with the initial description.

However, an easier way to manage things would be to define the AC to say we only show the feature tour for sites which currently do have Analytics setup and switch over to the GA4 Dashboard View. We could manage that by simply saying "don't show the tour if Analytics has just been setup", which could tie into the existing client-side state for when we show the Analytics setup success banner.

Please could you let me know which direction you'd prefer to take for the AC here?

**Comment by marrrmarrr** [comment]
@techanvil thanks for the detailed outline! Since this tour is only really relevant immediately after the dashboard is switched from UA to GA4, would it be simpler to limit the ACs only to this case? So we basically have two potential triggers for this tour: 
1) User clicks the ["Update dashboard" CTA](https://www.figma.com/file/0UCIE3hPeiu4NYdpAODASd/GA4-Updates?node-id=826-7063&t=wsvpNXFVSYAO2IDc-4) and their dashboard is reloaded with GA4 stats --> display tour
2) (post October 1) The dashboard is automatically switched over to GA4, [this banner](https://www.figma.com/file/0UCIE3hPeiu4NYdpAODASd/GA4-Updates?node-id=581-8816&t=wsvpNXFVSYAO2IDc-4) is displayed, and the user clicks "Learn what's new" CTA --> display tour

Perhaps connecting to these two events is easier than checking if the site is setting up GA for the first time?

**Comment by techanvil** [comment]
Thanks @marrrmarrr, that makes a lot of sense. I've updated the AC to cover the first point, and the second will be handled in subsequent issues  https://github.com/google/site-kit-wp/issues/6558 & https://github.com/google/site-kit-wp/issues/6549.

**Comment by aaemnnosttv** [comment]
@marrrmarrr what about the case for a secondary admin? The dashboard view setting is not user-specific so anyone who saw the dashboard before in UA mode wouldn't see the tour if we only showed it to the user who clicked the banner to switch the mode. The tour would also be missed in the case that a user manually switched the mode in the module settings.

**Comment by marrrmarrr** [comment]
@aaemnnosttv good catch, thanks for highlighting these corner cases. I've added comments on each below but tl;dr I'm not super concerned, as we won't be completely leaving anyone uninformed - the "NEW" badges also point to the updated metrics and provide more info + links on hover, so while they're not as conspicuous as the feature tour, they'll provide similar info.  

Re:switching from settings -- I'm expecting the huge majority of users to switch the view from the CTAs in the actual dashboard due to their prominent location, and potentially use the Settings toggle to flip back and forth (if at all necessary). So for that case in particular I think it's more beneficial to not show them the tour again and again every time they flip the Settings toggle. 

For secondary admins, are you referring to users who saw the dashboard via dashboard sharing, or who set up SK in parallel (without dashboard sharing)? Depending on the definition, I will dig in to see if I can track down how many such users we have. In any case I don't think this analysis should block us from proceeding with this issue. Maybe we can open a follow-up issue just for secondary admins?

**Comment by aaemnnosttv** [comment]
> I'm expecting the huge majority of users to switch the view from the CTAs in the actual dashboard due to their prominent location, and potentially use the Settings toggle to flip back and forth (if at all necessary). So for that case in particular I think it's more beneficial to not show them the tour again and again every time they flip the Settings toggle.

@marrrmarrr yes, for sure! Users only see feature tours once until they're dismissed so I'm not suggesting we'd change that. I suppose if a user were changing the dashboard view from the settings they probably know what they're doing and wouldn't need to see the tour.


> For secondary admins, are you referring to users who saw the dashboard via dashboard sharing, or who set up SK in parallel (without dashboard sharing)? Depending on the definition, I will dig in to see if I can track down how many such users we have. In any case I don't think this analysis should block us from proceeding with this issue.

By secondary admin, I mean another authenticated admin than the current one who is invoking the change of dashboard view from the banner notification. In this situation, they would end up on a different version of the dashboard (since the setting applies to all users), but wouldn't get the tour if it were only triggered by the banner as called for here. This scenario is similar to the one mentioned above where we show people a different banner after folks are automatically switched to the GA4 version of the dashboard, just a different cause.

So perhaps the requirements here already align with that – the tour would only be triggered automatically on the initial switch, and we could adjust #6558 to cover secondary admins too. What do you think?

**Comment by marrrmarrr** [comment]
@aaemnnosttv that is a great solution! That way they'll see the banner too and be more informed. 
Btw I checked and it looks like only 3% of sites have more than one admin connected currently.

**Comment by techanvil** [comment]
Hi @marrrmarrr and @aaemnnosttv, thanks for your collaboration on this!

Just so I'm clear on what's being proposed here, my understanding is this:

- Keep the AC for this issue as currently specced, i.e. change the conditions for showing the GA4 Reporting feature tour so it is only triggered when the user clicks the ["Update dashboard" CTA](https://www.figma.com/file/0UCIE3hPeiu4NYdpAODASd/GA4-Updates?node-id=826-7063&t=wsvpNXFVSYAO2IDc-4).
- Update the spec for issue https://github.com/google/site-kit-wp/issues/6558 to:
  - Include a version of the banner which has copy for the case where the user is a secondary admin and the dashboard view has been changed to GA4 by the primary admin, and adjust the conditions for showing the banner to include this case.
  - Actually render the new banner component, rather than rendering it as part of #6549.
  - Bring #6558 into the list of launch blockers (removing the _[~Oct 1]_ qualifier from the title).

@marrrmarrr, please could you confirm whether this is correct? Assuming so, would you mind providing the copy for the secondary admin version of the banner? Here's a screenshot of it as currently specced for reference.

![image](https://user-images.githubusercontent.com/20284937/217543128-c6340e31-6111-4ad2-ac1d-8585fb8dd3e4.png)

**Comment by marrrmarrr** [comment]
Hey @techanvil, thanks for the precise questions! 
Keeping the ACs of this issue as-is SGTM. 

For the secondary admin case, I think it might be helpful if @aaemnnosttv clarifies the different options here so that we cover everything -- e.g. it can be that the primary admin has not switched the dashboard view to GA4, but instead it's switched on automatically for all admins because of the timing (Oct 1). Does this make a difference in how this is implemented? 
Agree that we should cover the case of a primary admin switching over the dashboard view to GA4 in between feature launch and Oct 1 - in that case, it would be good if the secondary admin sees this banner, so they can see the feature tour as well. So we should implement at least the "Secondary admin" version of this banner as part of the feature development now. 

The other case (displaying the banner to users who didn't switch over even 3 months after UA stopped recording stats) is not as urgent in terms of time, so we could take care of it also after launching this feature.

**Comment by techanvil** [comment]
Thanks @marrrmarrr!

As we've also been discussing this on Slack I'll post a quick summary:

- Copy-wise, we'll keep the banner mostly as-is; rather than trying to come up with different wording for the various scenarios, we'll simply make it date-dependent, so "stopped" becomes "will stop" prior to July 1st.
- I've already got the Oct 1st version of banner developed locally so there's no real cost in pushing both versions up now (with the copy change described above the differences are minimal anyway).

**Comment by aaemnnosttv** [comment]
> For the secondary admin case, I think it might be helpful if @aaemnnosttv clarifies the different options here so that we cover everything -- e.g. it can be that the primary admin has not switched the dashboard view to GA4, but instead it's switched on automatically for all admins because of the timing (Oct 1). Does this make a difference in how this is implemented?

I think the cases for the banner can be conditional largely be based on the same logic we'd use to show the feature tour, although we might additionally check that UA is connected here (otherwise they'd only have the one version of the dashboard anyways). Otherwise, if the dashboard is in the GA4 mode and the user hadn't seen the GA4 dashboard tour yet, we'd show the banner. If the user dismisses the banner, we'd then also dismiss the tour which would prevent the banner from showing up again for that user too.

If the dashboard view was switched over due to the timing, I think the same logic would apply and the conditions for showing the banner could probably be the same. Again, I think this scenario is largely the same as if another admin changed the dashboard view, the mode just changed due to a manual change rather than automatically.

Since this issue will keep the AC as-is, it sounds like we're mostly discussing a new direction for #6558?

**Comment by marrrmarrr** [comment]
Thanks @aaemnnosttv ! Yeah, let's keep discussing in #6558

**Comment by techanvil** [comment]
Thanks, both - sounds good, I'll move this to IB and start work on it as this issue also blocks #6909.

**Comment by eugene-manuilov** [comment]
IB ✔️

**Comment by wpdarren** [comment]
### QA Update: ⚠️ 

@eugene-manuilov  While this passes the QAB, I do want to confirm that the tooltip tour takes around 30 seconds to appear after clicking on the CTA. This is because the `Top content` widget takes some time to load.  The conversions data loads straight away and I was expecting the tooltip tour to kick off at this point.  Interested in your thoughts on this. 

https://user-images.githubusercontent.com/73545194/235454331-759100a5-9ab3-48c6-9efc-09837061a8de.mp4

**Comment by eugene-manuilov** [comment]
@wpdarren we have #6909 that should make it load faster. For this ticket we just need to verify that the feature tour loads when you click on the CTA button.

**Comment by wpdarren** [comment]
### QA Update: ✅ 

**Verified:**

Clicking on the CTA button switches the dashboard to GA4 and the GA4 reporting feature tour starts.  As mentioned above there is a delay to the loading of the tour because the `Top content` widget takes some time to load but that will be fixed in 6909, so have ignored this issue while testing this.

**Comment by aaemnnosttv** [comment]
LGTM 👍
